### PR TITLE
Migrate site infrastructure to Antora

### PR DESCRIPTION
**Pull request by vy**
apache/logging-log4j2#2427 migrated Log4j 2 to Antora. apache/logging-log4j2#2443 cloned this work for Log4j 3. This PR clones the very same work to `logging-parent` to streamline it for all projects extending from it. A short summary of deliverables are shared below.

### Deliverables

* Website is migrated to Antora
* Auto-generated release notes is re-worked to avoid cluttering the Git history. Now it only runs during the `pre-site` phase and outputs to `target/generated-site` instead. That is, compiled `src/changelog` will not be dumped to `src/site` anymore.
* Auto-generated `constants.adoc` is directed to `target/generated-site` too (similar to release notes explained above)
* No web server is needed to view the generated website anymore, one can instead simply open `target/site` in a browser

### Implementation notes

Due to backward incompatible changes, the major version is bumped to 11.

### Review kit

1. Start with `pom.xml` changes
1. Review changes to the website, i.e., `src/site/antora/modules/ROOT`
1. Review changes to `.github`

### Future work

Once this feature gets released, `pom.xml` of Log4j 2 and 3 can be trimmed significantly.

### CODE REVIEW

File: .github/workflows/deploy-release-reusable.yaml [hunk 1]
```diff
@@ -28,10 +28,6 @@ on:
         description: Identifier used in the distribution artifact and Subversion repository folder filenames (e.g., `logging-parent`)
         required: true
         type: string
-      site-enabled:
-        description: Flag indicating if a website should be generated prior to deployment using `./mvnw site`
-        default: false
-        type: boolean
```
Comment (vy) [inline .github/workflows/deploy-release-reusable.yaml:34]: @ppkarwasz, note this decision too. If you want to make a release for a project, the project _must_ have a website too. This assumption helps with generating the release notes, which is used in several places; distribution ZIP, email templates, etc.

File: .github/workflows/deploy-release-reusable.yaml [hunk 2]
```diff
@@ -264,18 +280,16 @@ jobs:
           # Checkout the SVN repository
           export SVN_DIR="/tmp/svn-repo"
           svn co \
-            --depth files \
-            "https://dist.apache.org/repos/dist/dev/logging" \
+            "https://dist.apache.org/repos/dist/dev/logging/$PROJECT_ID" \
```
Comment (vy) [inline .github/workflows/deploy-release-reusable.yaml:283]: @ppkarwasz, note the assumption here: `PROJECT_ID` folder (i.e., `logging-parent`, `log4j`, `log4jt-tools`, etc.) must exist. The amount of boilerplate needed to deal with the intricacies of Subversion, and caching, and whatnot is frustrating; hence this simplification assumption.

File: src/site/antora/.antora.yml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+#   ███ ███  ██   ██ ██   ██ ██   ████ ██ ██   ████  ██████  ██
+#
+#  THIS FILE IS USED AS A TEMPLATE TO AUTO-GENERATE `antora.yml`!
+#
+#  The actual `antora.yml` that is used in `target/generated-site/antora` is auto-generated from `.antora.yml`.
+#  Auto-generation happens during `pre-site` phase of Maven.
+#  Hence, you must always
+#
+#  1. Edit `.antora.yml`
+#  2. Edit `antora.yml` to match the fields in `.antora.yml` and fill Maven property placeholders with dummy values
```
Comment (ppkarwasz) [inline src/site/antora/.antora.yml:32]: Please don't use files starting with a `.`. There is nothing to hide here.

Can you rename this to `antora.yml.tpl` or even `_antora.yml`?

**Review by ppkarwasz — APPROVED** [review]
LGTM

File: .github/workflows/build-reusable.yaml [hunk 4]
```diff
@@ -69,9 +69,34 @@ jobs:
...
+          path: |
+            node
+            node_modules
+
       - name: Build the website
         if: inputs.site-enabled
         shell: bash
+        env:
+          # Making Node.js cache hit visible for debugging purposes
+          NODEJS_CACHE_HIT: ${{ steps.nodejs-cache.outputs.cache-hit }}
```
[the remark this answers is not in the document]
Comment (ppkarwasz) [reply .github/workflows/build-reusable.yaml:99]: LGTM

File: .github/workflows/deploy-release-reusable.yaml (hunk 2 above)
Comment (ppkarwasz) [reply .github/workflows/deploy-release-reusable.yaml:283]: I don't think this is a problem.

**Comment by ppkarwasz** [comment]
Since all this PR does is modifying Github Actions can you stage the website after merging this PR to confirm that it works as required?

**Comment by vy** [comment]
> Since all this PR does is modifying Github Actions can you stage the website after merging this PR to confirm that it works as required?

I will do that in my follow up PR, where I will implement automatic website publication. :wink:

**Pull request merged**